com.microsoft.mrtk.core/Utilities/CameraFOVChecker.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit
+{
+    /// <summary>
+    /// Camera extension methods to test if colliders are within camera's FOV. Uses caching to improve performance and
+    /// ensure values are only computed once per frame.
+    /// </summary>
+    public static class CameraFOVChecker
+    {
+        private static int inFOVLastCalculatedFrame = -1;
+#if !NETFX_CORE
+        // Map from grabbable => is the grabbable in FOV for this frame. Cleared every frame
+        private static Dictionary<ValueTuple<Collider, Camera>, bool> inFOVColliderCache = new Dictionary<ValueTuple<Collider, Camera>, bool>();
+#else
+        private static Dictionary<Tuple<Collider, Camera>, bool> inFOVColliderCache = new Dictionary<Tuple<Collider, Camera>, bool>();
+#endif
+        // List of corners shared across all sphere pointer query instances --
+        // used to store list of corners for a bounds. Shared and static
+        // to avoid allocating memory each frame
+        private static List<Vector3> inFOVBoundsCornerPoints = new List<Vector3>();
+
+        /// <summary>
+        /// Returns true if a collider's bounds is within the camera FOV.
+        /// <para/>
+        /// Utilizes a cache to test if this collider has been seen before and returns current frame's calculated result.
+        /// </summary>
+        /// <param name="myCollider">The collider to test.</param>
+        /// <remarks>
+        /// NOTE: This is a 'loose' FOV check -- it can return true in cases when the collider is actually not in the FOV
+        /// because it does an axis-aligned check when testing for large colliders. So, if the axis aligned bounds are in
+        /// the bounds of the camera, it will return true.
+        /// </remarks>
+        public static bool IsInFOVCached(this Camera cam, Collider myCollider)
+        {
+            // If the collider's size is zero, it is not visible. Return false.
+            if (myCollider.bounds.size == Vector3.zero || myCollider.transform.localScale == Vector3.zero)
+            {
+                return false;
+            }
+
+#if !NETFX_CORE
+            ValueTuple<Collider, Camera> cameraColliderPair = ValueTuple.Create(myCollider, cam);
```

Comment:
Also, now that Unity 2020 is our min version, we can update all the `ValueTuple` references to `(Collider, Camera)` if we want :) so this could be

`(Collider, Camera) cameraColliderPair = (myCollider, cam);`

Reply:
done :)